### PR TITLE
docs: add pages to describe workspace retention rules

### LINKED ISSUE #134: docs: add documentation about workspace retention rules
Tasks:

- add a page that describes how to use retention rules work from a user perspective
- add a page describing how to set up retention rules for administrators.
- update `reana.yaml` specification page

### PR DESCRIPTION
**Pull request by VMois**
- add reference to reana.yaml page;
- add page describing retention rules to users;
- add page describing how to setup retention rules to administrators.

closes #134

### CODE REVIEW

**Review by audrium — DISMISSED** [review]
Looks good to me! Well written and clear

File: docs/administration/configuration/configuring-workspace-retention-rules/index.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Configuring workspace retention rules
```
Comment (tiborsimko) [inline docs/administration/configuration/configuring-workspace-retention-rules/index.md:1]: I would call it "Configuring global workspace retention rules" or some such as we called it in the user-facing documentation, to indicate that this is not for users, but for admins.

(Also, I think this section is perhaps too long; I'll propose some shortening it in the comments below.)

File: docs/administration/configuration/configuring-workspace-retention-rules/index.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+REANA allows users to keep files in their workflows for a limited time using **retention rules**.
```
Comment (tiborsimko) [inline docs/administration/configuration/configuring-workspace-retention-rules/index.md:3]: As of version 0.9, REANA administrators can configure a global workflow file retention period after which all the files produced by user workflows that are not declared as "inputs" or "outputs" will be automatically deleted after a certain number of days. (By default, every file is kept indefinitely.)

If you would like to limit the workspace file retention period globally, you can configure workspaces.retention_rules.maximum_period ` Helm value to the desired amount of days, such as 3650. This will make sure that all files produced by users will be kept at most for ten years. Note that users will be able to define their individually desired retention rules (link) to a period not exceeding this global maximum.

We recommend to keep the `workspaces.retention_rules.maximum_period` value to its default value `forever` and  rather use the user disk storage resource quotas (link) to limit the amount of storage that each user can consume. (This is the default REANA behaviour.) 

Note that if you redeploy REANA and modify its `workspaces.retention_rules.maximum_period` settings between two rolling deployments, e.g. from "forever" to "3650 days", the new workflows will use the updated global maximum retention period (ten years) while the old workflows will keep using older default (forever).

File: docs/advanced-usage/workspace-retention-rules/index.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# Workspace retention rules
+
+There could be situations where you would like to keep some files in your workflow's workspace only for a limited time.
```
[the remark this answers is not in the document]
Comment (mdonadoni) [reply docs/advanced-usage/workspace-retention-rules/index.md]: Some little changes:
- disk quota resource limits -> disk quota limit
- GitLab asset expiry configuration -> GitLab job artifacts expiry

File: docs/administration/configuration/configuring-workspace-retention-rules/index.md (hunk 2 above)
Comment (mdonadoni) [reply docs/administration/configuration/configuring-workspace-retention-rules/index.md]: I have removed `after a certain number of days` to avoid the repetition with `... after which ...`.
I have also removed `(This is the default REANA behaviour.)` as we were already saying `... its default value ...`.

File: docs/advanced-usage/workspace-retention-rules/index.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+## What happens when a workflow is restarted?
+
+When you restart a workflow, the current and the previous runs of the workflow share the same workspace.
+REANA will only consider the retention rules of the latest restart and it will ignore rules from earlier runs.
+However, please note that some retention rules might already have been applied if the workflow is restarted after their expiration.
+
+## Limitation on the maximum numbers of retention days
+
+Administrators of the REANA cluster can set a maximum retention period for the entire workspace depending on the storage needs, as described in [this documentation page](/administration/configuration/configuring-workspace-retention-rules).
```
[the remark this answers is not in the document]
Comment (mdonadoni) [reply docs/advanced-usage/workspace-retention-rules/index.md]: I have removed `that you can set` from the first sentence, to avoid the repetition of `can set`

File: docs/advanced-usage/workspace-retention-rules/index.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
+# Workspace retention rules
+
+There could be situations where you would like to keep some files in your workflow's workspace only for a limited time.
+This can be useful, for example, to free up storage space to stay within your [quota limits](/advanced-usage/user-quotas).
+REANA allows you to do so by defining custom **retention rules** in the `reana.yaml` specification file of your workflow.
+
+## Defining retention rules
```
[the remark this answers is not in the document]
Comment (mdonadoni) [reply docs/advanced-usage/workspace-retention-rules/index.md]: I kept the full title "Defining workspace file retention rules" even though I changed the title of the page to the shorter version. WDYT?

File: docs/advanced-usage/workspace-retention-rules/index.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    data/*.json: 30
+...
+```
+
+Each rule consists of two parts:
+
+- *pattern* (e.g. `tmp/*.root`) that specifies which files and directories are affected;
+- *duration* (e.g. `7`) that indicates for how many *days* to keep the files after the end of the workflow.
+
+For instance, considering the example above, the rule `tmp/*.root: 7` means "*when the workflow ends, delete files and directories whose name ends with `.root` in the `tmp` directory after 7 days*".
```
[the remark this answers is not in the document]
Comment (mdonadoni) [reply docs/advanced-usage/workspace-retention-rules/index.md]: I prefer keeping the additional explanation

File: docs/advanced-usage/workspace-retention-rules/index.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+!!! note
+    REANA will not apply the retention rules to files and directories specified in the *inputs* and *outputs* of `reana.yaml`.
+    This ensures that your workflows can be reproduced even after applying the retention rules.
+    If you no longer need inputs and outputs, consider deleting the files or the workflow manually.
+
+Here you can find a table of all available patterns to match files and directories:
+
+| Pattern   | Meaning                                                  |
+| --------- | -------------------------------------------------------- |
+| `*`       | match everything                                         |
```
[the remark this answers is not in the document]
Comment (mdonadoni) [reply docs/advanced-usage/workspace-retention-rules/index.md]: I have also better specified that `[seq]` and `[!seq]` match only a single character

File: docs/advanced-usage/workspace-retention-rules/index.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+| `**`      | match this directory and all subdirectories, recursively |
+| `?`       | match any single character                               |
+| `[seq]`   | match any character in `seq`                             |
+| `[!seq]`  | match any character *not* in `seq`                       |
```
[the remark this answers is not in the document]
Comment (mdonadoni) [reply docs/advanced-usage/workspace-retention-rules/index.md]: The table contains only the basic wildcard that can be used to build more complex pattern, so I prefer keeping it as is. I have added the explanation about `mystage[12]/mydata*.csv` in the paragraph below the table

File: docs/advanced-usage/workspace-retention-rules/index.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+```yaml
+workspace:
+  retention_days:
+    "**/*": 30
+```
+
+## What happens when a workflow is restarted?
+
+When you restart a workflow, the current and the previous runs of the workflow share the same workspace.
+REANA will only consider the retention rules of the latest restart and it will ignore rules from earlier runs.
```
[the remark this answers is not in the document]
Comment (mdonadoni) [reply docs/advanced-usage/workspace-retention-rules/index.md]: I have also removed the second "therefore" to avoid the repetition

**Review by tiborsimko — COMMENTED** [review]
Looks nice! Some more comments after rereading the page, left here to ease identification:

- In order to make the reading look less scary, I would start with a new tiny paragraph describing generous defaults first. This would give four small introductory paragraphs instead of current two:

```
REANA usually holds all your workflow workspace files "forever", i.e. until you actively decide to delete them. 

However, there could be situations when ...

REANA allows you to configure...

This feature is similar...
```

- "Defining workspace file retention rules" -> "Defining custom workspace file retention rules " to underline that people don't have to do this if they don't want to.

- "Retention rules are specified" -> "A custom retention rules for your workflow runs can be specified..." to reinfornce the same message.

- "However, please note that" -> I would actually delete this phrase. It interrupts the flow for me, and since we never delete declared inputs/outputs, people may wonder why it is necessary to add such a phrase.  If we keep it, then I would rephrase as follows: "However, please note that some of the previous retention rules might have already been applied if the workflow was restarted after their application time. You may be especially careful if you modify your workflow run inputs or outputs in between restarts." 

- "Limits on the maximum number of retention days" -> Perhaps add a phrase just after the example saying "The value None means that there is no theoretically imposed maximum limit on the retention days you can use, which effectively means that you can declare some workspace files to be retained "forever", just like your workflow inputs and outputs are."

- "You can verify the retention period settings for your workflow run" -> "You can verify the custom retention period settings for any of your workflow runs"

- "by means of `reana-client`" -> "by means of `reana-client` command-line client:"

**Comment by mdonadoni** [comment]
> * "However, please note that" -> I would actually delete this phrase. [...]

Removed 

> * "Limits on the maximum number of retention days" -> Perhaps add a phrase just after the example saying "The value None means that there is no theoretically imposed maximum limit on the retention days you can use, which effectively means that you can declare some workspace files to be retained "forever", just like your workflow inputs and outputs are."

I have added the phrase up to `... on the retention days you can use`, as there isn't a way to declare files to be retained forever, except by not defining any retention rules matching them.

**Review by tiborsimko — APPROVED** [review]
